applications/vanilla/models/class.categorymodel.php

```diff
@@ -659,6 +659,12 @@ public static function JoinUserData(&$Categories, $AddUserCategory = TRUE) {
          $Categories[$CID]['PermsDiscussionsEdit'] = $Session->CheckPermission('Vanilla.Discussions.Edit', TRUE, 'Category', $Category['PermissionCategoryID']);
          $Categories[$CID]['PermsCommentsAdd'] = $Session->CheckPermission('Vanilla.Comments.Add', TRUE, 'Category', $Category['PermissionCategoryID']);
       }
+
+      // Translate name and description
+      foreach ($IDs as $ID) {
+         $Categories[$ID]['Name'] = TranslateContent($Categories[$ID]['Name']);
+         $Categories[$ID]['Description'] = TranslateContent($Categories[$ID]['Description']);
```

Comment:
I'm a bit concerned here. Picture a long sentence description that you translate and then you want to change that description. All of a sudden your translation breaks. I'm wondering if the following would be better:

```
$Code = $Categories[$ID]['UrlCode']
$Categories[$ID]['Name'] = TranslateContent("Categories.$Code.Name", $Categories[$ID]['Name']);
$Categories[$ID]['Description'] = TranslateContent("Categories.$Code.Description", $Categories[$ID]['Description']);
```